### ISSUE TITLE
Followup for #18331 "build: Use git archive as source tarball"

### ISSUE THREAD

**Issue by dongcarl** [bitcoin/bitcoin#18547]
After the merge of #18331, there are now two ways to create source tarballs.

1. `make dist`
2. `git archive`

It would seem that both methods are being used in our Gitian descriptor.

First, we run `make dist` to create a bootstrapped tarball: https://github.com/bitcoin/bitcoin/blob/c5966a87d1fdd7a98f2baee5b2deddd541fdfb5a/contrib/gitian-descriptors/gitian-linux.yml#L143-L148

Which is used by each host to build the binary tarballs in the for loop:
https://github.com/bitcoin/bitcoin/blob/c5966a87d1fdd7a98f2baee5b2deddd541fdfb5a/contrib/gitian-descriptors/gitian-linux.yml#L154-L191

Then, we use `git archive` to create the final source tarball: https://github.com/bitcoin/bitcoin/blob/c5966a87d1fdd7a98f2baee5b2deddd541fdfb5a/contrib/gitian-descriptors/gitian-linux.yml#L194

It seems to me that perhaps it'd be good to reinstate the behaviour whereby our binary tarballs are built from the source tarball, just for robustness's sake.

A simple fix might be:
1.  switch the earlier `make dist` to `git archive`
2. `./autogen.sh` for each host
3. Remove the last call to `git archive`

Perhaps this can even allow us to specify the `no-dist` automake option and clean up `dist` related targets in `Makefile.am`.

**Comment by dongcarl** [comment]
So if I'm reading https://github.com/bitcoin/bitcoin/pull/18349#issuecomment-607404517 correctly, we can't ditch `make dist`?

I feel somewhat defeated on this topic :disappointed:. It seems to me that the result of all this fiddling is:

1. We have now introduced a second way of producing source tarballs without being able to replace the first. We have to maintain both ways.
2. We've changed the relationship between our source tarballs and binary tarballs in that the binary ones are no longer being built from the source ones.
